### PR TITLE
(FACT-2140) Added Options class, refactored FactManager

### PR DESCRIPTION
**Pull request by sebastian-miclea**

### CODE REVIEW

File: lib/framework/core/fact_manager.rb [hunk 1]
```diff
@@ -13,9 +12,8 @@ def initialize
     end
 
     def resolve_facts(options = {}, user_query = [])
```
Comment (oanatmaria) [inline lib/framework/core/fact_manager.rb]: maybe cli_options would be a better name

File: lib/framework/core/options/default_options.rb [hunk 2]
```diff
@@ -10,16 +10,20 @@ def augment_with_defaults!
...
 
     def cli_defaults
-      @options[:debug] = false if @options[:debug].nil?
-      @options[:trace] = false if @options[:trace].nil?
-      @options[:verbose] = false if @options[:verbose].nil?
-      @options[:log_level] = 'error' unless @options[:log_level]
+      @options[:debug] = false
+      @options[:trace] = false
+      @options[:verbose] = false
+      @options[:log_level] = 'warn'
```
Comment (oanatmaria) [inline lib/framework/core/options/default_options.rb]: this should be symbol

Comment (IrimieBogdan) [reply lib/framework/core/options/default_options.rb]: Replaced with symbol.

File: lib/framework/core/fact_manager.rb (hunk 1 above)
Comment (IrimieBogdan) [reply lib/framework/core/fact_manager.rb]: Changed to cli_options

**Review by oanatmaria — APPROVED** [review]

**Pull request merged**